src/adapter.js

```diff
@@ -157,15 +165,22 @@ function makeSourceMap(SourceMap, ast, element, path) {
 }
 
 // Make a new annotation for the given path and message
-function makeAnnotation(Annotation, SourceMap, ast, result, info, path, message) {
+function makeAnnotation(Annotation, Link, SourceMap, ast, result, info, path, message) {
   const annotation = new Annotation(message);
   annotation.classes.push(info.type);
   annotation.code = info.code;
   result.content.push(annotation);
 
+  if (info.relation) {
+    const link = new Link();
+    link.relation = 'origin';
+    link.href = `http://docs.apiary.io/validations/swagger#${info.relation}`;
+    annotation.links.push(link);
```

Comment:
**Out of scope question—beware**

Could we improve the API here? 

```js
const link = annotations.links.add('link'); // automatically initializes and pushes to content
link.relation = 'origin';
link.href = `http://docs.apiary.io/validations/swagger#${info.relation}`;
```

We could then have a list of registered elements to add to content for arrays. Seems very common.